db/migrate/20210224190247_migrate_shipping_taxes.rb

```diff
@@ -0,0 +1,83 @@
+class MigrateShippingTaxes < ActiveRecord::Migration
+  def up
+    return unless instance_uses_shipping_tax?
+
+    create_shipping_tax_rates
+    assign_to_shipping_methods
+    migrate_tax_amounts_to_adjustments
+  end
+
+  def instance_uses_shipping_tax?
+    Spree::Preference.find_by(key: '/spree/app_configuration/shipment_inc_vat').value
```

Comment:
What if instances change this? They have to create the tax rates themselves?

Reply:
Yes, so instances that already apply taxes to shipments will have the equivalent tax rates created and applied where they are currently used (but with actual tax rates and categories instead of a single global value). Instances that don't currently apply a tax rate will still be able to apply any tax category they like to shipments.

Most instances actually already have all relevant tax rates set up per category, like "10% GST", so if they don't use shipping taxes already they can apply those if they want or create new ones.